src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs

```diff
@@ -0,0 +1,254 @@
+
+        let stack_owner_id = owner_id.as_did_str().to_stack_string();
+
+        let dataset_entry_rows = sqlx::query_as!(
+            DatasetEntryRowModel,
+            r#"
+            SELECT dataset_id   as "id: _",
+                   owner_id     as "owner_id: _",
+                   dataset_name as name,
+                   created_at   as "created_at: _"
+            FROM dataset_entries
+            WHERE owner_id = $1
+            "#,
+            stack_owner_id.as_str(),
+        )
+        .fetch_all(connection_mut)
+        .await
+        .map_int_err(GetDatasetEntriesByOwnerIdError::Internal)?;
+
+        Ok(dataset_entry_rows.into_iter().map(Into::into).collect())
+    }
+
+    async fn save_dataset_entry(
+        &self,
+        dataset_entry: &DatasetEntry,
+    ) -> Result<(), SaveDatasetEntryError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr
+            .connection_mut()
+            .await
+            .map_err(SaveDatasetEntryError::Internal)?;
+
+        let stack_dataset_id = dataset_entry.id.as_did_str().to_stack_string();
+        let stack_owner_id = dataset_entry.owner_id.as_did_str().to_stack_string();
+
+        sqlx::query!(
+            r#"
+            INSERT INTO dataset_entries(dataset_id, owner_id, dataset_name, created_at)
+                VALUES ($1, $2, $3, $4)
+            "#,
+            stack_dataset_id.as_str(),
+            stack_owner_id.as_str(),
+            dataset_entry.name.as_str(),
+            dataset_entry.created_at,
+        )
+        .execute(connection_mut)
+        .await
+        .map_err(|e| match e {
+            sqlx::Error::Database(e) if e.is_unique_violation() => {
+                let postgres_error_message = e.message();
+                if postgres_error_message.contains("idx_dataset_entries_owner_id_dataset_name") {
+                    DatasetEntryNameCollisionError::new(dataset_entry.name.clone()).into()
+                } else {
+                    SaveDatasetEntryErrorDuplicate::new(dataset_entry.id.clone()).into()
+                }
+            }
+            _ => SaveDatasetEntryError::Internal(e.int_err()),
+        })?;
+
+        Ok(())
+    }
+
+    async fn update_dataset_entry_name(
+        &self,
+        dataset_id: &DatasetID,
+        new_name: &DatasetName,
+    ) -> Result<(), UpdateDatasetEntryNameError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr
+            .connection_mut()
+            .await
+            .map_err(UpdateDatasetEntryNameError::Internal)?;
+
+        let stack_dataset_id = dataset_id.as_did_str().to_stack_string();
+
+        let update_result = sqlx::query!(
+            r#"
+            UPDATE dataset_entries
+                SET dataset_name = $2
+                WHERE dataset_id = $1
+            "#,
+            stack_dataset_id.as_str(),
+            new_name.as_str(),
```

Comment:
Minor: reorder, so as not to catch the eye
```sql 
SET dataset_name = $1
WHERE dataset_id = $1
```

Reply:
Corrected here and in SQlite original